optimum/intel/openvino/configuration.py

```diff
@@ -542,29 +535,63 @@ def post_init(self):
-            self.weight_format = "int4" if self.bits == 4 else "int8"
-        if self.weight_format not in ["int4", "int8", "mxfp4", "nf4"]:
+        if self.dtype is None:
+            self.dtype = "int4" if self.bits == 4 else "int8"
+        if self.dtype not in ["int4", "int8", "mxfp4", "nf4"]:
             raise ValueError(
-                f"Weight format must be one of the following: ['int4', 'int8', 'mxfp4', 'nf4'], but found: {self.weight_format}."
+                f"Weights quantization data type must be one of the following: ['int4', 'int8', 'mxfp4', 'nf4'], but found: {self.dtype}."
             )
-        if self.weight_format in ["mxfp4", "nf4"]:
+        if self.dtype in ["mxfp4", "nf4"]:
             if self.bits != 4:
                 raise ValueError(
-                    f"When applying weight compression with '{self.weight_format}' weight format, the `bits` parameter must be set to 4, but found {self.bits}"
+                    f"When applying weight compression with '{self.dtype}' data type, the `bits` parameter must be set to 4, but found {self.bits}"
                 )
-            if self.weight_format == "mxfp4":
+            if self.dtype == "mxfp4":
                 if self.quant_method == OVQuantizationMethod.AWQ:
-                    raise ValueError("The AWQ algorithm is not supported for 'mxpf4' weight format")
+                    raise ValueError("The AWQ algorithm is not supported for 'mxpf4' data type")
                 if self.scale_estimation:
-                    raise ValueError("The Scale Estimation algorithm is not supported for 'mxpf4' weight format")
+                    raise ValueError("The Scale Estimation algorithm is not supported for 'mxpf4' data type")
                 if self.gptq:
-                    raise ValueError("The GPTQ algorithm is not supported for 'mxfp4' weight format")
+                    raise ValueError("The GPTQ algorithm is not supported for 'mxfp4' data type")
                 if self.lora_correction:
-                    raise ValueError("The LoRA Correction algorithm is not supported for 'mxfp4' weight format")
+                    raise ValueError("The LoRA Correction algorithm is not supported for 'mxfp4' data type")
         if self.gptq and self.lora_correction:
             raise ValueError("The GPTQ and LoRA Correction algorithms can't be applied simultaneously")
 
+    def to_nncf_dict(self) -> Dict[str, Any]:
+        """
+        Returns a dictionary with the variables that are ready to use for nncf.quantize() call.
+        """
+
+        signed_bitness = {4: "int4", 8: "int8"}
+        mode = self.dtype if self.dtype else signed_bitness[self.bits]
+        if mode in signed_bitness.values():
+            mode += "_sym" if self.sym else "_asym"
+        if mode == "mxfp4":
+            mode = "e2m1"
+        mode = nncf.CompressWeightsMode(mode)
+
+        awq = None
+        if self.quant_method == "awq" or self.quant_method == OVQuantizationMethod.AWQ:
```

Comment:
I believe that `self.quant_method` is an instance of `OVQuantizationMethod`, according to https://github.com/huggingface/optimum-intel/blob/f61b7e86264c2719610338a56ac1b971c6047d2a/optimum/intel/openvino/configuration.py#L442

Reply:
Removed